### PR TITLE
improve documentation of Similarity surface

### PR DESCRIPTION
**Pull request by videlec**
Fix issue https://github.com/videlec/sage-flatsurf/issues/21

### CODE REVIEW

**Comment by videlec** [comment]
@slel you can have a look (solves #21)

File: flatsurf/geometry/similarity_surface.py [hunk 1]
```diff
@@ -46,19 +46,57 @@ class SimilaritySurface(SageObject):
...
+        sage: Stop.set_edge_pairing(0, 3, 1, 1)
+        sage: Stop.set_edge_pairing(1, 2, 2, 1)
+        sage: Stop.set_edge_pairing(1, 0, 2, 3)
+        sage: S = SimilaritySurface(Stop)
+        sage: S
+        SimilaritySurface built from 3 polygons
+
+        sage: TestSuite(S).run()    # important sanity check!
+
+    Finally, you can als implement a similarity surface by inherting from
```
Comment (slel) [inline flatsurf/geometry/similarity_surface.py]: als -> also, inherting -> inheriting

File: flatsurf/geometry/similarity_surface.py [hunk 2]
```diff
@@ -46,19 +46,57 @@ class SimilaritySurface(SageObject):
...
+    :class:`SimilaritySurface` and implement the methods:
+
+    - ``base_ring(self)``: the base ring in which coordinates lives
+
+    - ``polygon(self, lab)``: the polygon associated to the label ``lab``
+
+    - ``base_label(self)``: return a first label
+
+    - ``opposite_edge(self, lab, edge)``: a couple (``other_label``,
```
Comment (slel) [inline flatsurf/geometry/similarity_surface.py]: couple -> pair

File: flatsurf/geometry/similarity_surface.py [hunk 3]
```diff
@@ -46,19 +46,57 @@ class SimilaritySurface(SageObject):
...
+        sage: TestSuite(S).run()    # important sanity check!
+
+    Finally, you can als implement a similarity surface by inherting from
+    :class:`SimilaritySurface` and implement the methods:
+
+    - ``base_ring(self)``: the base ring in which coordinates lives
+
+    - ``polygon(self, lab)``: the polygon associated to the label ``lab``
+
+    - ``base_label(self)``: return a first label
```
Comment (slel) [inline flatsurf/geometry/similarity_surface.py]: "return a first label" -> "which label to use as the base one"

File: flatsurf/geometry/similarity_surface.py [hunk 4]
```diff
@@ -46,19 +46,57 @@ class SimilaritySurface(SageObject):
...
+
+    - ``opposite_edge(self, lab, edge)``: a couple (``other_label``,
+      ``other_edge``) representing the edge being glued
+
+    - ``is_finite(self)``: return whether the surface is built from finitely many polygons
```
Comment (slel) [inline flatsurf/geometry/similarity_surface.py]: "return whether" -> "whether"

File: flatsurf/geometry/similarity_surface.py [hunk 5]
```diff
@@ -46,19 +46,57 @@ class SimilaritySurface(SageObject):
     label of the polygons is done at startup. If the set is finite then by
     default the labels are the first non-negative integers 0,1,...
 
-    The edge are identified by a couple (polygon label, edge number).
+    The edges are identified by a pair (polygon label, edge number).
```
Comment (slel) [inline flatsurf/geometry/similarity_surface.py]: identified -> labeled

**Comment by slel** [comment]
Added a few comments inline.

Should the documentation mention that all polygon sides must
be glued (i.e. no way to define a surface with boundary here)?

File: flatsurf/geometry/similarity_surface.py (hunk 1 above)
Comment (videlec) [reply flatsurf/geometry/similarity_surface.py]: thanks, fixed in 0dcd491

File: flatsurf/geometry/similarity_surface.py (hunk 2 above)
Comment (videlec) [reply flatsurf/geometry/similarity_surface.py]: also fixed in 0dcd491

File: flatsurf/geometry/similarity_surface.py (hunk 3 above)
Comment (videlec) [reply flatsurf/geometry/similarity_surface.py]: also fixed in 0dcd491

File: flatsurf/geometry/similarity_surface.py (hunk 5 above)
Comment (videlec) [reply flatsurf/geometry/similarity_surface.py]: also fixed in 0dcd491

File: flatsurf/geometry/similarity_surface.py (hunk 4 above)
Comment (videlec) [reply flatsurf/geometry/similarity_surface.py]: also fixed in 0dcd491

**Comment by videlec** [comment]
> Should the documentation mention that all polygon sides must
> be glued (i.e. no way to define a surface with boundary here)?

You can define the surface (no check is made). But the test suite will fail (documented after 21a8524).

**Comment by slel** [comment]
Thanks!

**Pull request merged**